src/deployNodeApp.js

```diff
@@ -367,48 +377,79 @@ async function deployNodeApp (packageJson /*: Object */, env /*: string */, opts
+      )
+    }
+    tags.hash = `${tags.image}:${newTag}`
+  }
 
   if (opts.push) {
-    if (!answers.confirmed) {
-      if (answers.registry.includes('index.docker.io')) {
-        process.stdout.write(
-          `${chalk.yellow(
-            '!!'
-          )} You are using Docker Hub. If the docker repository does not exist,\n` +
-            `   it may be automatically created with ${chalk.yellow('PUBLIC')} access!\n` +
-            '   Make sure you have all secrets in your ".dockerignore" file,\n' +
-            '   and you may want to make sure your image repository is setup securely!\n\n'
-        )
-      }
+    if (answers.registry.includes('index.docker.io')) {
+      process.stdout.write(
+        `\n${chalk.yellow('!!')} You are using Docker Hub. If the repository "${
+          tags.image
+        }" does not exist, it may be automatically created with ${chalk.yellow('PUBLIC')} access!
+${chalk.yellow(
+    '!!'
+  )} You can visit https://cloud.docker.com/repository/create to create a ${chalk.yellow(
+  'private repository'
+)} instead.
+${chalk.yellow('!!')} In any case, make sure you have all secrets in your ".dockerignore" file.\n\n`
+      )
     }
+  }
 
-    if (!answers.confirmed && opts.confirm && opts.output !== '-') {
-      const { confirmed } = await inquirer.prompt([
-        {
-          name: 'confirmed',
-          type: 'confirm',
-          message: 'Are you sure you want to continue?'
-        }
-      ])
-      if (!confirmed) {
-        process.exit(1)
+  if (typeof answers.isPublic !== 'boolean' && (opts.confirm && opts.output !== '-')) {
+    const { isPublic } = await inquirer.prompt([
+      {
+        name: 'isPublic',
+        type: 'list',
+        message: 'Is this a public or private repository?',
+        choices: [
+          { key: 'P', name: 'Private (recommended)', value: false },
+          { key: 'U', name: 'Public', value: true }
+        ],
+        default: 0
       }
-      answers.confirmed = confirmed
-    }
+    ])
+    answers.isPublic = isPublic
   }
 
-  if (!packageJson['deploy-node-app']) packageJson['deploy-node-app'] = {}
-  packageJson['deploy-node-app'][env] = answers
+  if (answers.isPublic === false) {
+    const regcred = await execSyncWithEnv(
+      `kubectl --context=${answers.context} get secret ${appName}-regcred || echo "no"`
+    )
+    if (regcred === 'no') {
+      const { password } = await inquirer.prompt({
+        name: 'password',
+        type: 'password',
+        message: 'What is your docker hub password?',
+        validate: function (password) {
+          if (password.length <= 1) return 'Invalid password'
+          return true
+        }
+      })
+      await execSyncWithEnv(
+        `kubectl --context=${answers.context} \
+          create secret docker-registry ${appName}-regcred \
+          --docker-server=https://${answers.registry} \
+          --docker-username=${answers.registryUsername} \
+          --docker-password=${password}`
```

Comment:
If `password` (or, really, any of these values) contains characters that the shell interprets on its own, this will fail.  For example, a password of `ab&cd` will die because the command will be interpreted as `kubectl...--docker-password=ab` run in the background, followed by `cd`.

Reply:
Ah, excellent catch - will fix. This is a sort of embarrassing one but in my defense it was very late at night :P

Reply:
No worries - if it hadn't have been for my randomly-generated Docker hub password, I wouldn't have caught it either.  ;)

Reply:
This should now be fixed in fea6a6c :D